src/with_ladj.jl

```diff
@@ -81,12 +81,22 @@ end
 
 _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj::Tuple{Any,Real}) = y_with_ladj
```

Comment:
To fix ambiguity errors:
```suggestion
function _with_ladj_on_mapped(map_or_bc::F, y_with_ladj::Tuple{Any,Real})  where {F<:Union{typeof(map),typeof(broadcast)}}
    return y_with_ladj
end
```